Lib/unittest/case.py

```diff
@@ -328,18 +307,39 @@ class _AssertLogsContext(_BaseTestCaseContext):
     def __init__(self, test_case, logger_name, level):
         _BaseTestCaseContext.__init__(self, test_case)
         self.logger_name = logger_name
+        import logging
+        self.logging = logging
         if level:
             self.level = logging._nameToLevel.get(level, level)
         else:
             self.level = logging.INFO
         self.msg = None
 
     def __enter__(self):
+        logging = self.logging
         if isinstance(self.logger_name, logging.Logger):
             logger = self.logger = self.logger_name
         else:
             logger = self.logger = logging.getLogger(self.logger_name)
         formatter = logging.Formatter(self.LOGGING_FORMAT)
+
+        class _CapturingHandler(logging.Handler):
```

Comment:
I dislike defining a new _CapturingHandler class each time _AssertLogsContext context manager is used :-( Maybe a lazy import could be used, something similar to PR #19600?